core/controllers/reader.py

```diff
@@ -629,6 +665,58 @@ class ExplorationCompleteEventHandler(base.BaseHandler):
     The state name recorded should be a state with a terminal interaction.
     """
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': editor.SCHEMA_FOR_EXPLORATION_ID
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.ENTITY_ID_REGEX
+                    }]
+                },
+                'default_value': None
+            },
+            'version': {
+                'schema': editor.SCHEMA_FOR_VERSION
+            },
+            'state_name': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'has_length_at_most',
+                        'max_value': constants.MAX_STATE_NAME_LENGTH
+                    }]
+                }
+            },
+            'session_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'client_time_spent_in_secs': {
+                'schema': {
+                    'type': 'float',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 0
+                    }]
+                }
+            },
+            'params': {
+                'schema': {
+                    'type': 'dict',
+                    'properties': []
+                }
```

Comment:
Please add an explanatory comment, why are params set like this.

Reply:
I have navigated both the frontend and backend parts and I found that they are customizable for any exploration (they cannot be used in newer explorations). So, Probably the type should be `dict` here.

Reply:
1. Later I found `params` handler arg in the domain layer, which means we should choose the type `object_dict`. 
2. Then by your suggestion I got to know what params is. I found in the backend that it follows dict[str, str]  type and I took `validation_method` to add rules for it.